test/message/test_plan.py

```diff
@@ -948,3 +948,33 @@ def plan_double_abs(x):
     assert isinstance(ptr_result.child, sy.PointerTensor)
     result = ptr_result.get()
     assert th.equal(result, expected)
+
+
+def test_cached_plan_send(workers):
+    bob = workers["bob"]
+
+    @sy.func2plan(args_shape=[(1,)])
+    def plan_abs(data):
+        return data.abs()
+
+    plan_bob_ptr1 = plan_abs.send(bob)
+    plan_bob_ptr2 = plan_abs.send(bob)
```

Comment:
Check also that plan_bob_ptr1 and plan_bob_ptr2 are the same object :) [maybe the id / id_at_location, etc]